### ISSUE TITLE
Initializing vue tables drives me crazy

### ISSUE THREAD

**Issue by decebal** [matfish2/vue-tables#51]
- Vue.js version: "vue": "^1.0.24",
- consumed using: webpack
- FULL error message (including stack trace):

`
server.created	@	funeral-plans.9b70455….min.js:formatted:8069
Vue._callHook	@	funeral-plans.9b70455….min.js:formatted:1933
Vue._init	@	funeral-plans.9b70455….min.js:formatted:665
Vue._init	@	funeral-plans.9b70455….min.js:formatted:5943
VServerTable	@	VM803:2
build	@	funeral-plans.9b70455….min.js:formatted:3970
mountComponent	@	funeral-plans.9b70455….min.js:formatted:3939
(anonymous function)	@	funeral-plans.9b70455….min.js:formatted:3918
(anonymous function)	@	funeral-plans.9b70455….min.js:formatted:3930
cb	@	funeral-plans.9b70455….min.js:formatted:137
Vue._resolveComponent	@	funeral-plans.9b70455….min.js:formatted:2067
resolveComponent	@	funeral-plans.9b70455….min.js:formatted:3932
setComponent	@	funeral-plans.9b70455….min.js:formatted:3917
bind	@	funeral-plans.9b70455….min.js:formatted:3908
Directive._bind	@	funeral-plans.9b70455….min.js:formatted:4253
linkAndCapture	@	funeral-plans.9b70455….min.js:formatted:1399
compositeLinkFn	@	funeral-plans.9b70455….min.js:formatted:1385
Vue._compile	@	funeral-plans.9b70455….min.js:formatted:1979
Vue.$mount	@	funeral-plans.9b70455….min.js:formatted:2352
Vue._init	@	funeral-plans.9b70455….min.js:formatted:666
Vue._init	@	funeral-plans.9b70455….min.js:formatted:5943
App	@	VM804:2
(anonymous function)	@	funeral-plans.9b70455….min.js:formatted:6907
_match	@	funeral-plans.9b70455….min.js:formatted:6916
onChange	@	funeral-plans.9b70455….min.js:formatted:6728
listener	@	funeral-plans.9b70455….min.js:formatted:6454
start	@	funeral-plans.9b70455….min.js:formatted:6458
start	@	funeral-plans.9b70455….min.js:formatted:6808
(anonymous function)	@	funeral-plans.9b70455….min.js:formatted:7530
__webpack_require__	@	init.410acd6….min.js:1
(anonymous function)	@	funeral-plans.9b70455….min.js:formatted:4
__webpack_require__	@	init.410acd6….min.js:1
webpackJsonpCallback	@	init.410acd6….min.js:1
(anonymous function)	@	funeral-plans.9b70455….min.js:formatted:1
`
- relevant code:
`

    import VueTables from 'vue-tables'
    import Vue from 'vue'
    Vue.use(VueTables.server, {filterByColumn: false, compileTemplates: true});

    export default {
        name: "App",
        el: "#table",
        props: {
            columns:{
                type: Array,
                required:true
            },
            data: {
                type: Array,
                required: true
            }
        },
        data: function() {
            return {
                columns: ['name', 'phone_number', 'email_address', 'vendor', 'created_at', 'status'],                
            };
        }
    }

        <div class="col-md-6 col-md-offset-3" id="#table">
            <v-server-table v-ref:table :columns="columns" :options="options"></v-server-table>
        </div>
`
- steps for reproducing the issue:

I swear I got this working before, but I might have lost a dependency or something on the road and can't find a solution to this:

`Uncaught TypeError: Cannot read property '0' of undefined :  this.initOrderBy(this.columns[0])`

**Comment by matfish2** [comment]
Why are you passing props to a non-component? It seems like your `columns` prop interferes with the `columns` array that you are returning.
Also, you need to pass a `url` prop to the server component.

**Comment by decebal** [comment]
sorry @matfish2 , I removed that because I saw no relevance in it. You are right all of thm should be in there and they are in my code.

It's the loading in webpack that's failing and I don't know why or how to reproduce it, just thought that this footprint would suffice if you encountered the problem by yourself.

I also tried the basic examples from the Readme file all of them had an issue in the "created" function, with the "options" property.

Thank you for responding so quick. 
Do you think you need some more details on this ?

webpack loaders are `vue` and `vue-html`

**Comment by matfish2** [comment]
I didn't encounter it myself because I am using `browserify`. So you'll have to figure that one out yourself.

The last error you included - which is the one I was referring to - is not a compiler error, but rather a run-time one. It suggests that your `columns` prop is undefined.

**Comment by decebal** [comment]
Alright so the problem was is the first piece of code I submitted, not sure why it happened, but this is my understanding:

if I defined the prop "columns" without a default it would throw that error, also it would not call the "data" function while having the props defined.

Solution 1: I removed the whole "props"  entry from my component definition and it worked.
Solution 2: I moved the logic from "data" to "props" defaults and it worked.

Which one would be recommended ?
